src/bamboo.rs

```diff
@@ -0,0 +1,156 @@
+use crate::{
+	error,
+	Result,
+};
+use curl::easy::Easy;
+use futures::stream::FuturesUnordered;
+use rayon::prelude::*;
+use serde::*;
+use serde::{
+	Deserialize,
+	Serialize,
+};
+use snafu::ResultExt;
+use std::collections::HashMap;
+use std::io::{
+	stdout,
+	Write,
+};
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct EmployeesDirectoryResponse {
+	employees: Vec<EmployeesDirectoryInnerResponse>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct EmployeesDirectoryInnerResponse {
+	id: String,
+	display_name: Option<String>,
+	first_name: Option<String>,
+	last_name: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct EmployeeResponse {
+	id: String,
+	#[serde(rename = "customRiotID")]
+	riot_id: Option<String>,
+	#[serde(rename = "customGithub")]
+	github: Option<String>,
+}
+
+fn get_employees_directory(access_token: &str) -> Result<EmployeesDirectoryResponse> {
+	let mut dst = Vec::new();
+	let mut handle = Easy::new();
+	handle
+		.url("https://api.bamboohr.com/api/gateway.php/parity/v1/employees/directory")
```

Comment:
I think we should pull the base url out into a constant and then concat the relative directories, in case we ever need to change it for some reason. We can also set the URL at runtime if we wanted to allow for a bot to be able to run on any bamboo instance.

```rust
fn endpoint(url: &str) -> String {
   const BAMBOO_API_URL: &str = "https://api.bamboohr.com/api/gateway.php/parity/v1/";
   let mut s = String::from(BAMBOO_API_URL);
   s += url;
   s
}

handle.url(Self::endpoint("employees/directory"))
```